praw/models/reddit/modmail.py

```diff
@@ -0,0 +1,46 @@
+"""Provide models for new modmail."""
+from ...const import API_PATH, urlparse
+from ...exceptions import ClientException
+from .base import RedditBase
+
+
+class ModmailConversation(RedditBase):
+    """A class for modmail conversations."""
+
+    STR_FIELD = 'id'
+
+    @staticmethod
+    def id_from_url(url):
+        """Return the ID contained within a conversation URL.
+
+        :param url: A url to a conversation in the following format:
+            * https://mod.reddit.com/mail/all/2gmz
+
+        Raise :class:`.ClientException` if URL is not a valid conversation URL.
+
+        """
+        parsed = urlparse(url)
+        if not parsed.netloc:
+            raise ClientException('Invalid URL: {}'.format(url))
+
+        submission_id = parsed.path.rsplit('/', 1)[-1]
+
+        if not submission_id.isalnum():
+            raise ClientException('Invalid URL: {}'.format(url))
+        return submission_id
+
+    def __init__(self, reddit, id=None,  # pylint: disable=redefined-builtin
+                 url=None, _data=None):
+        """Construct an instance of the ModmailConversation object."""
+        super(ModmailConversation, self).__init__(reddit, _data)
+
+        if id is not None:
+            self.id = id  # pylint: disable=invalid-name
+        elif url is not None:
+            self.id = self.id_from_url(url)
+
+    def _fetch(self):
```

Comment:
That sounds like the best approach once the parsing is there, but right now I'm getting a problem where, because `objectify` is just returning the raw data as a `dict`, `RedditBase._fetch` fails when it tries to access the `__dict__` attribute of the result of `Reddit.get`. Thoughts?